include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -845,7 +845,9 @@ namespace internal
         return accessor.reference_cell().standard_vs_true_line_orientation(
           pair[1],
           face_orientation_raw(accessor, quad_index),
-          accessor.quad(quad_index)->line_orientation(line_within_face_index));
+          static_cast<unsigned char>(
+            accessor.quad(quad_index)
+              ->line_orientation(line_within_face_index)));
```

Comment:
This is a `bool` to `unsigned char` conversion, right? I'm not sure this is correct here -- I think that we need to call a conversion function that does a more complicated conversion. So the existing code my be a bug, and the explicit conversion just papers over it. I would suggest to just taking this part of the patch out.

Reply:
It looks like this is easy enough to fix here so I pushed a corresponding commit.